### ISSUE TITLE
Stopped working a few days ago with Recived an error: undefined

### ISSUE THREAD

**Issue by evandrocoan** [backstrokeapp/server#102]
It stopped creating pull requests automatically, then, I tried to sync manually:
![image](https://user-images.githubusercontent.com/5332158/50351074-335bdd80-0528-11e9-9936-a450bd3f5fd6.png)

Every repository I try to sync does this `Recived an error: undefined`

If I try to access the direct URL under `Webhook link`:
![image](https://user-images.githubusercontent.com/5332158/50351189-8c2b7600-0528-11e9-813f-0a56a3dcce2c.png)

I got this json reponse:
`
{"error":"EXECABORT Transaction discarded because of previous errors.","stack":"ReplyError: EXECABORT Transaction discarded because of previous errors.\n    at parseError (/app/node_modules/redis-parser/lib/parser.js:193:12)\n    at parseType (/app/node_modules/redis-parser/lib/parser.js:303:14)"}
`

**Comment by alanfx** [comment]
Also seeing this issue

**Comment by 1egoman** [comment]
I don't have the bandwidth to maintain Backstroke anymore.

It's a project that has done fairly well, and many people use it and find it helpful. But, it's a project that does too much for a single person to maintain, and it's got a number of moving parts that all have a tendency to break at the worst time. For a while, I found it fun to maintain, but over the past year it's turned into a chore, and I don't really have an interest in keeping it running.

Normally, when people shut things down, an alternative is usually provided. Backstroke doesn't really have any alternatives that I know of (though if one exists and seem decent, I'll gladly edit this post with a link), so I'd say that if Backstroke is really something you need, the best "upgrade path" I can think of and what I'd recommend is to set up a small server that either listens for a webhook (like backstroke classic) or polls the main branch on an upstream for changes. Then, use the Github API to create a new pull request. Backstroke classic, an example of this sort of approach, can be found here: https://github.com/backstrokeapp/legacy

If anyone has an interest in either taking over the project, please let me know. I'd be more than willing to pass it off to you and walk you through setting it up for development. If there isn't any interest, I'll likely shut it off later in the year once Backstroke's sponsored digitalocean credit runs out.

Sorry for this being so abrupt. To be honest, I've been thinking about doing this for a while and I should have been a bit more honest in mentioning this prior to something breaking. I'm also going to put this notice in the readme to make this more clear.

- Ryan, @1egoman

**Comment by alanfx** [comment]
Hey @1egoman, I really appreciate your work on Backstroke, and I understand the demands of developing a project "on the side". I'm wondering if you could publicly document the steps for setting up a development environment and deployment of the new code base, so that potential users or maintainers could setup Backstroke servers for their use and then also continue developing the code. I'd much prefer to try to move the new codebase forward, instead of using the legacy version. Would that be possible?

**Comment by Deadlyelder** [comment]
@1egoman Indeed, as @alanfx suggested, it would be great if you could document the development and deployment configurations it would be great. I too would be interested in this.

**Comment by 1egoman** [comment]
@alanfx @Deadlyelder Yes, I can do that. Some documentation already exists here, but it could probably be improved: https://github.com/backstrokeapp/docs

**Comment by alanfx** [comment]
@1egoman Thanks a lot for agreeing to do this, Ryan. I have looked at those docs, but it wasn't very clear to me how to get it all up and running, so a more step-by-step document would help a lot.

**Comment by 1egoman** [comment]
@alanfx Ah, I sent you the wrong repository, sorry (been a while since I touched this stuff!) - what I really meant was https://github.com/backstrokeapp/deployment. That should be pretty close to a step by step document for setting up Backstroke in development. It doesn't give an exact configuration on how to deploy Backstroke into a more "production" environment, but it does explain how one would modify the development configuration and it leaves it to the reader as an exercise.

If that document isn't sufficient, I can answer any questions you or others may have but in general it should be a good primer.

**Comment by evandrocoan** [comment]
I think this is an alternative/similar solution:
1. https://github.com/wei/pull a GitHub App built with probot that keeps your repository up-to-date with upstream changes via automated PRs.

**Comment by 1egoman** [comment]
@evandrocoan Yes, this has been mentioned in the README.

**Comment by evandrocoan** [comment]
Sorry, you updated the README.md one day after I read it.

Why closing this issue? Just because you are not fixing it, does not mean is should be closed. Otherwise, you still missing to close all the other 7 issues which still opened. May be consider archiving this repository? https://help.github.com/articles/archiving-a-github-repository/

I just looked over the other bot, and I would call it a very dangerous bot (https://github.com/wei/pull/issues/76) because by default it will override you fork contents with the upstream and needs a lot of work in order to setup it without erasing stuff. Differently from backstroke, which would just create a pull request.

I would assume/continue backstroke maintenance, but after a some years using and getting pull requests notifications popping up all the time, I decided to setup a local script which I run only when I got the patience to.

I had setup backstroke with about 200 forks and with that much, I got a fairly share of notification popping up almost all the time. A very annoying experience. The worst times were when:
1. Got pull request for the Repository A at 20:00 o'clock, then I merge it
1. Now, it is 21:00 o'clock, I got another pull request notification about the same Repository A. Then, I merge it again.
1. Now, it is 23:00 o'clock, I got again another pull request notification for the same Repository A. Then, Now merge it again.
1. And this thing keeps going for the Repository A
1. This work flow became very annoying.

In the future, I may think about setting a server with backstroke or that other dangerous bot (after making it non-dangerous) and deal with these minor caveats. For now, I will try to create a local script which I can launch, when I fell like merging a bunch of pull requests. For reference, I am starting setup this script here: https://github.com/evandrocoan/githubpullrequests

I plan on working on that script for the next days and adding a few new stuff like:
1. A report review listing all forks I own, but are not listed to open pull requests. This way I can easy see/spot/find some repository I may had forgotten to setup for pull requests.
1. A command to merge all pull requests which does not have any conflicts.
1. Report how may pull requests it created and for which repositories, and how many pull requests failed to be created and why.
